### DISCUSSION TITLE
Is it worth calling `dispatcher.close()` for resource deallocation?

### DISCUSSION THREAD

**Discussion by schaable** [nodejs/undici#3522] [Q&A] [answered]
I’m trying to understand the role of `dispatcher.close()` in Undici. Specifically, I have a setup where multiple `Pool` instances are created, and I’m wondering if it's a good idea to call `dispatcher.close()` when I no longer need a Pool instance.

Could you expand on what dispatcher.close() does and if it's necessary to call it to properly deallocate resources?

**Comment by mcollina** [comment, marked as answer]
Yes it is. It will not cause a leak long term, but it will 100% keep some resources allocated until all sockets are terminated.
